astroid/inference.py

```diff
@@ -832,3 +835,17 @@ def instance_getitem(self, index, context=None):
             node=self, index=index, context=context))
 
 bases.Instance.getitem = instance_getitem
+
+
+def _set_explicit_context_for_call(call, context):
```

Comment:
One problem I have with this is that it mutates the underlying nodes with a context. While we do that for the explicit inference, I'd like us to start considering the nodes as immutable, with as few as possible places where we'd have to mutate them (right now another one apart of `explicit_inference` would be mutating `.locals`). It would make more sense to push these values inside the context itself, after which we could modify the inference to look for the nested values of a context when inferring values. This way we keep nodes untouched, we only pass the modified context.